pkg/operator/csidriveroperator/driver_starter.go

```diff
@@ -218,6 +230,14 @@ func (dsrc *driverStarterCommon) sync(ctx context.Context, syncCtx factory.SyncC
 				return err
 			}
 			if !shouldRun {
+				// If shouldRun is false, then the
```

Comment:
This branch will be executed for all CSI drivers that do not belong to the current platform, i.e. for most of these: https://github.com/openshift/cluster-storage-operator/blob/813fcbe37e3ad47521601763641999fec920cefa/pkg/operator/operator_starter.go#L203-L214

You can check if this loop actually matched any CSI driver operator and set `Upgradeable=true` only when it did not.

Or, you can always unconditionally create one `StorageOperatorrUpgradeable=true`. If any ClusterCSIDriver then reports `*Upgradeable=False`, [ClusterOperatorStatusController](https://github.com/openshift/cluster-storage-operator/blob/813fcbe37e3ad47521601763641999fec920cefa/pkg/operator/operator_starter.go#L105) will do an union where one `*Upgradeable=false` trumps any number of `*Upgradeable=true`.

Reply:
Thanks! I moved the `setUpgradeableTrue` call to after the loop, if none of these controllers have (ever) started.

I'm hesitant to set Upgradeable=true unconditionally because then in the most common case, where at least one controller is started, we may introduce a brief window of time where it's set to true but will soon change to false. Maybe this is benign, but I would like to avoid that flip-flop and set it only if we know those controllers will not run.